src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Concatenate.cs

```diff
@@ -38,6 +38,69 @@ public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures(int arity)
             return base.GetSignatures(arity);
         }
 
+        public override bool CheckInvocation(TexlBinding binding, TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        {
+            Contracts.AssertValue(args);
+            Contracts.AssertValue(argTypes);
+            Contracts.Assert(args.Length == argTypes.Length);
+            Contracts.Assert(args.Length >= 2);
+            Contracts.AssertValue(errors);
+
+            int count = args.Length;
+            bool fArgsValid = true;
+            nodeToCoercedTypeMap = null;
+
+            for (int i = 0; i < count; i++)
+            {
+                var typeChecks = CheckType(args[i], argTypes[i], DType.String, errors, SupportCoercionForArg(i), out DType coercionType);
+                if (typeChecks && coercionType != null)
+                    CollectionUtils.Add(ref nodeToCoercedTypeMap, args[i], coercionType);
+
+                fArgsValid &= typeChecks;
+            }
+
+            if (!fArgsValid)
+                nodeToCoercedTypeMap = null;
+
+            returnType = ReturnType;
+
+            return fArgsValid;
+        }
+    }
+
+    // Concatenate(source1:s|*[s], source2:s|*[s], ...)
+    // Corresponding DAX function: Concatenate
+    // Note, this performs string/table, table/table, table/string concatenation, but not string/string
+    // Tables will be expanded to be the same size as the largest table. For each scalar, a new empty table
+    // will be created, and the scalar value will be used to fill the table to be the same size as the largest table
+    internal sealed class ConcatenateTableFunction : BuiltinFunction
+    {
+        public override bool IsSelfContained => true;
+        public override bool SupportsParamCoercion => true;
+
+        public ConcatenateTableFunction()
+            : base("Concatenate", TexlStrings.AboutConcatenate, FunctionCategories.Table | FunctionCategories.Text, DType.EmptyTable, 0, 2, int.MaxValue)
+        { }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
+        {
+            yield return new[] { TexlStrings.ConcatenateArg1, TexlStrings.ConcatenateArg1 };
+            yield return new[] { TexlStrings.ConcatenateArg1, TexlStrings.ConcatenateArg1, TexlStrings.ConcatenateArg1 };
+            yield return new[] { TexlStrings.ConcatenateArg1, TexlStrings.ConcatenateArg1, TexlStrings.ConcatenateArg1, TexlStrings.ConcatenateArg1 };
+        }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures(int arity)
+        {
+            if (arity > 2)
+                return GetGenericSignatures(arity, TexlStrings.ConcatenateArg1, TexlStrings.ConcatenateArg1);
+            return base.GetSignatures(arity);
+        }
+
+        public override string GetUniqueTexlRuntimeName(bool isPrefetching = false)
```

Comment:
[minor] do we need to specify the default value for the parameter if it's not used?

Reply:
Yes because this is an override.